### DISCUSSION TITLE
Inconsistent Cropping Behavior

### DISCUSSION THREAD

**Discussion by benlubas** [3rd/image.nvim#38] [Q&A]
I just have a question about the way cropping works when an image is too big to fit where it's placed.

I feel like I've seen two behaviors:
1. The image gets cropped and you only see part of it, but it's aspect ratio stays the same, but part of it is missing
2. the image gets shrunk, and you can still see the entire thing, but it get's smaller and smaller

number two is the behavior that I've seen more recently, and I'm wondering what has changed, is that a kitty thing? or something that the plugin is doing? I see that the kitty backend in this plugin has the "crop" feature, so I'm guessing that this is something kitty is doing, and the other behavior was the way image magick was cropping images.

I do prefer behavior 1, where it just shows part of the image and I have to scroll to see the rest. Is there currently a way to configure the plugin to use that behavior with kitty?

**Comment by 3rd** [comment]
It should follow the max width/height settings, I think there are some bugs around that.
Option 1 should work when there's no configured max size, and option 2 should work otherwise.
I'll run some tests and check how it behaves!

**Reply by benlubas** [reply]
Now that you say that that's totally how it works! Thanks for clearing that up!

**Reply by benlubas** [reply]
@3rd I've done some testing, and I noticed that when an image is bound to a window it will be cropped rather than shrunk when there is max size configured, but I'm calling render with an image geometry that's equal to the max size.

<details>
<summary>Examples</summary>

image in a floating window:

![image](https://github.com/3rd/image.nvim/assets/56943754/d08e7c06-269b-402b-a68e-23cd03b4aa1d)

same image with same geometry, but bound to the current window:

![image](https://github.com/3rd/image.nvim/assets/56943754/6fe5e099-ec59-40c4-8f0a-f5ce77901ff6)

and for comparison's sake, same image, same geometry, no window at all:

![image](https://github.com/3rd/image.nvim/assets/56943754/f1410ad7-8851-4ea2-a52e-876dbc014089)

</details> 
this is causing problems in molten. I'm trying to render a floating window that's only as tall as it needs to be to show all the content in it. Images are fine when there is no configured max_width or max_height and the max height percent is set to 100. However, when I set a max width of 10 in the image.nvim config, and then in molten account for that by rendering this image with a geometry that will not be changed by the max height (ie. this image is rendered with a `img:render({...height = 10, width = &lt;computed width&gt;`), the image is still cropped and I can't see the entire thing. 

![image](https://github.com/3rd/image.nvim/assets/56943754/2d14efb9-dff9-4758-8890-abceda7e9057)

**Reply by 3rd** [reply]
Got it, will look into it!
